src/snowflake/cli/plugins/snowpark/package/commands.py

```diff
@@ -131,46 +133,127 @@ def package_upload(
     """
-    if _deprecated_install_option:
-        install_packages = _deprecated_install_option
+    allow_shared_libraries_pypi_option = {
+        True: PypiOption.YES,
+        False: PypiOption.NO,
+    }[allow_shared_libraries]
+    if deprecated_allow_native_libraries != PypiOption.NO:
+        allow_shared_libraries_pypi_option = deprecated_allow_native_libraries
+
+    if ignore_anaconda:
+        anaconda = None
+    else:
+        anaconda = AnacondaChannel.from_snowflake()
+        package = Requirement.parse(name)
+        if anaconda.is_package_available(
+            package, skip_version_check=skip_version_check
+        ):
+            return MessageResult(
+                f"Package {name} is already available in Snowflake Anaconda Channel."
+            )
 
-    lookup_result = lookup(
-        name=name,
-        install_packages=install_packages,
-        allow_native_libraries=allow_native_libraries,
+    packages_dir = SecurePath(".packages")
+    packages_are_downloaded, dependencies = download_packages(
+        anaconda=anaconda,
+        perform_anaconda_check_for_dependencies=not ignore_anaconda,
+        package_name=name,
+        requirements_file=None,
+        packages_dir=packages_dir,
+        index_url=index_url,
+        allow_shared_libraries=allow_shared_libraries_pypi_option,
+        skip_version_check=skip_version_check,
     )
 
-    if not isinstance(lookup_result, (NotInAnaconda, RequiresPackages)):
-        return MessageResult(lookup_result.message)
+    if not packages_are_downloaded:
+        return MessageResult(
+            dedent(
+                f"""
+                Cannot create package for {name}. Please check the package name
+                or try again with --allow-shared-libraries option.
+                """
+            )
+        )
 
-    # The package is not in anaconda so we have to pack it
+    # The package is not in anaconda, so we have to pack it
     zip_file = create_packages_zip(name)
     message = dedent(
         f"""
-    Package {zip_file} created. You can now upload it to a stage using
-    snow snowpark package upload -f {zip_file} -s <stage-name>`
-    and reference it in your procedure or function.
-    """
+        Package {zip_file} created. You can now upload it to a stage using
+        snow snowpark package upload -f {zip_file} -s <stage-name>`
+        and reference it in your procedure or function.
+        Remember to add it to imports in the procedure or function definition.
+        """
     )
-    if isinstance(lookup_result, RequiresPackages):
-        message += "\n" + lookup_result.message
+    if dependencies.snowflake:
+        message += dedent(
+            f"""
+            The package {name} is supported, but does depend on the
+            following Snowflake supported libraries. You should include the
+            following dependencies in you function or procedure requirements:
+            """
```

Comment:
That's not true, `snowflake!=None` means there are dependencies available in Anaconda that needs to be added to packages list because we do not include them in the package.

Basically we are operating on three information:
1. What packages are available in anaconda (dependencies.snowflake)
2. What packages needs to be packed (dependencies.other)
3. Does packages to be packed include shared code (we don't store this info)